include/fmt/std.h

```diff
@@ -242,6 +246,51 @@ struct formatter<std::optional<T>, Char,
 FMT_END_NAMESPACE
 #endif  // __cpp_lib_optional
 
+#ifdef __cpp_lib_expected
+FMT_BEGIN_NAMESPACE
+namespace detail {
+
+template <typename Char, typename OutputIt, typename T>
+auto write_expected_alternative(OutputIt out, const T& v) -> OutputIt {
+  if constexpr (has_to_string_view<T>::value)
+    return write_escaped_string<Char>(out, detail::to_string_view(v));
+  else if constexpr (std::is_same_v<T, Char>)
+    return write_escaped_char(out, v);
+  else
```

Comment:
nit: `else`s are redundant because of early returns, let's remove them

Reply:
https://github.com/fmtlib/fmt/blob/961d4dd8300ef5af4c71c73129f0c81cb53d4063/include/fmt/std.h#L343-L351

just an exact copy of the impl in variant.
We could also factor that out and use the impl in both.

Reply:
> We could also factor that out and use the impl in both.

Good idea, let's do it.